### PR TITLE
chore: Add target specification

### LINKED ISSUE #10943: Install target specification
We need a specification that outlines the requirements for all install targets. Covering aspects like deployment architecture, high availability, and hardening. It should be a clear checklist that all of our targets conform to, structuring the upcoming target audits to enable low friction deployments of Vector.

### PR DESCRIPTION
**Pull request by binarylogic**
Closes #10943

This adds the first bits for a target specification, which outlines requirements intended to align how we support various targets (AKA platforms). The intent is to represent the various recommendations made in the new "going to production" documentation, making it easy for users to deploy Vector in production environments.

### CODE REVIEW

File: docs/specs/target.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      48 GiB of SSD persisted disk space.
+* Scaling
+  * Load-balancing?...
+  * Autoscaling SHOULD be enabled by default driven by an 85% CPU utilization
+    target over a rolling 5 minute window.
+  * Autoscaling SHOULD have a stabilization period of 5 minutes.
+
+### 5. Unified Architecture
+
+TODO: Should we support this as a top-level architecture, or have users deploy
```
Comment (spencergilbert) [inline docs/specs/target.md:111]: As a top-level arch: This would be awkward to accomplish with the current chart design and possibly with upgrade processes down the line. I'd lean towards two deployments.

File: docs/specs/target.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    * Downloaded Vector artifacts MUST be verified against the provided
+      checksum.
+    * The latest Vector version SHOULD be downloaded unless otherwise specified
+      by the user.
+  * Vector's configuration
+    * Vector's configuration directory SHOULD be read restricted to Vector's
+      service account.
+  * Vector's runtime
+    * Vector MUST be run under an unprivileged, deciated service account by
+      default.
```
[the remark this answers is not in the document]
Comment (binarylogic) [reply docs/specs/target.md:139]: Yeah, we might consider splitting the spec on the deployment roles/architectures. I'm doing that with the docs now and it's helpful.

[the remark this answers is not in the document]
Comment (binarylogic) [reply docs/specs/target.md:139]: Then let's move it under the aggregator section and make it exclusive to that architecture.

**Comment by spencergilbert** [comment]
> align how we support various targets (AKA platforms)

@binarylogic I'm curious if that means we'll also be aligning this on the site rather than splitting the `Installation` block between package managers, platforms, OS's, and manual

**Comment by binarylogic** [comment]
I'm not sure what you mean specifically, but we'll still organize installation docs across the various targets. We want to meet the user at their desired installation process and then take them as far as we can within each deployment role. So each target's installation tutorial will be split across the deployment roles (agent and aggregator). Each target will be able to take the user farther into those roles depending on its capabilities. For example, the Helm charts will be able to take the user much further in deploying the aggregator than the the Debian package target. At the very least, each target can ship with the role's default configuration.

File: docs/specs/target.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
+# Target Specification
+
+This document specifies requirements for deployment targets for the
```
Comment (spencergilbert) [inline docs/specs/target.md:3]: Broadly we can split our targets between `Virtual/Physical Machines` and `Orchestrators`.

- `Virtual/Physical Machines`
  - Target is a single instance.
  - This implies limited, to no, _High Availability_ and _Scaling_ spec adherence.
  - However, there should be no issues with general _Architecture_, _Sizing_, and _Hardening_.
- `Orchestrators`
  - Target is a group of nodes, managed by a scheduler or orchestrator.
  - A management layer (Kubernetes, Nomad, Swarm, Mesos, DC/OS) over a group of nodes allows us to adhere to _High Availability_ and _Scaling_ spec more easily.
  - These targets should expose primitives to ensure the _Architecture_, _Sizing_, and _Hardening_ spec is met.

There is some space between these two classes for things like Chef, Puppet, Ansible, or even Terraform that could help bridge this gap but we do not currently provide any resources for these types of deployments.

File: docs/specs/target.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    - Vector's service account SHOULD NOT have the ability to overwrite Vector's
+      binary or configuration files. The only directory the Vector service
+      account should write to is Vector’s data directory.
+- Network hardening
+  - Configured sources and sinks SHOULD use encrypted channels by default.
+
+[agent_architecture]: https://vector.dev/docs/setup/going-to-prod/arch/agent/
+[aggregator_architecture]: https://vector.dev/docs/setup/going-to-prod/arch/aggregator/
+[default_agent_configuration]: https://github.com/vectordotdev/vector/blob/target-specification/config/agent/vector.yaml
+[default_aggregator_configuration]: https://vector.dev/docs/reference/configuration/default/aggregator.yaml
```
Comment (spencergilbert) [inline docs/specs/target.md:151]: Not sure where we want the default configs to live, or at least where to link to them.

File: docs/specs/target.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+[hardening]: https://vector.dev/docs/setup/going-to-prod/hardening/
+[high_availability]: https://vector.dev/docs/setup/going-to-prod/high-availability/
+[reference_architectures]: https://vector.dev/docs/setup/going-to-prod/arch/
+[RFC 2119]: https://datatracker.ietf.org/doc/html/rfc2119
+[terminology_document]: https://vector.dev/docs/reference/glossary/
```
Comment (spencergilbert) [inline docs/specs/target.md:156]: This is the most terminology document I could find that we publish, but it's lacking a number of definitions - specifically the mentioned `"agent", "aggregator", "node", and "service"`.

File: docs/specs/target.md (hunk 4 above)
Comment (binarylogic) [reply docs/specs/target.md:151]: Somewhere in the the `config` directory in the Vector repo. So `config/agent.yaml` and `config/aggregator.yaml`, or `config/defaults/*`/

File: docs/specs/target.md (hunk 5 above)
Comment (binarylogic) [reply docs/specs/target.md:156]: That's fine for now. Yes, it will be in the reference on the new site and updated.

File: docs/specs/target.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+### 4. Aggregator Architecture
+
+The [aggregator architecture][aggregator_architecture] deploys Vector onto
+dedicated nodes for data aggregation. Along with general [hardening](#6-hardening)
+requirements, the following requirements define support for this architecture:
+
+- Architecture
+  - SHOULD deploy as a service on dedicated nodes, Vector is the only service
+    on the node.
+  - MUST deploy with a persistent disk that is available between deployments.
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply docs/specs/target.md:99]: I think `SHOULD` is still good here. If users are using aggregators with something like Kafka or SQS than a disk is unnecessary. That is: I think we should still support Stateless Aggregators in k8s.

File: docs/specs/target.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+  - The Vector service SHOULD NOT be artificially limited with resource
+    limiters such as cgroups.
+  - SHOULD request 8 vCPUs, MUST be overridable by the user.
+  - SHOULD request 2 GiB of memory per vCPU (16 GiB in this case), MUST be
+    overridable by the user.
+  - SHOULD request 6 GiB of disk space per vCPU (48 GiB in this case), MUST be
+    overridable by the user.
+  - SHOULD use the following instances if deployed in a listed cloud.
+    - AWS SHOULD default to `c6g.2xlarge` instances with 48 GiB of EBS `io2`
+      disk space.
```
Comment (jszwedko) [inline docs/specs/target.md:121]:
```suggestion
    - AWS SHOULD default to `c6i.2xlarge` instances with 48 GiB of EBS `gp3`
      disk space.
```

I believe this is what we discussed?

File: docs/specs/target.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+    - Azure SHOULD default to `f8` instances with 48 GiB of standard SSD disk
+      space.
+    - GCP SHOULD default to `c2` instances with 8 vCPUS, 16 GiB of memory, and
+      48 GiB of SSD persisted disk space.
```
Comment (jszwedko) [inline docs/specs/target.md:125]: I assume we mean `pd-ssd` for the disk type here? Could we add that to be extra clear? We could link to https://cloud.google.com/compute/docs/disks#disk-types too.

File: docs/specs/target.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+  - SHOULD request 8 vCPUs, MUST be overridable by the user.
+  - SHOULD request 2 GiB of memory per vCPU (16 GiB in this case), MUST be
+    overridable by the user.
+  - SHOULD request 6 GiB of disk space per vCPU (48 GiB in this case), MUST be
+    overridable by the user.
+  - SHOULD use the following instances if deployed in a listed cloud.
+    - AWS SHOULD default to `c6g.2xlarge` instances with 48 GiB of EBS `io2`
+      disk space.
+    - Azure SHOULD default to `f8` instances with 48 GiB of standard SSD disk
+      space.
```
Comment (jszwedko) [inline docs/specs/target.md:123]:
```suggestion
    - Azure SHOULD default to `f8` instances with 48 GiB of Standard SSD disk
      space.
```

It wasn't clear to me that "Standard SSD" is an Azure term. We could link to https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#standard-ssds here too.

File: docs/specs/target.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+  - Vector's data directory SHOULD be read and write restricted to Vector's
+    dedicated service account.
+  - Core dumps SHOULD be prevented for the Vector process to prevent in flight
+    data from leaking to disk.
+- Process hardening
+  - Vector's artifacts
+    - All communication during the setup process, such as downloading Vector
+      artifacts, MUST use encrypted channels.
+    - Downloaded Vector artifacts MUST be verified against the provided
+      checksum.
```
Comment (jszwedko) [inline docs/specs/target.md:165]: Just noting we don't have these yet.

File: docs/specs/target.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+### 5. Aggregator Architecture
+
+The [aggregator architecture][aggregator_architecture] deploys Vector onto
+dedicated nodes for data aggregation. Along with general [hardening](#7-hardening)
+requirements, the following requirements define support for this architecture:
+
+- Architecture
+  - SHOULD deploy as a service on dedicated nodes, Vector is the only service
+    on the node.
```
Comment (jszwedko) [inline docs/specs/target.md:98]: I guess this is a SHOULD for a VM target, but not for k8s?

File: config/agent/vector.yaml [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
```
Comment (jszwedko) [inline config/agent/vector.yaml:1]: Should I review these default configuration files? They seem orthogonal to the target spec.

**Review by jszwedko — COMMENTED** [review]
Apologies for the delay in review! I think this is heading in the right direction. I left a few comments below.

File: docs/specs/target.md (hunk 7 above)
Comment (spencergilbert) [reply docs/specs/target.md:121]: This was awaiting the updated going-to-prod docs but I _believe_ those are right.

Comment (jszwedko) [reply docs/specs/target.md:121]: Gotcha, yeah, we can wait and see. I know we want to use `c6i` for now since `c6g` seemed to have worse performance and we don't have soak tests for it yet. I thought we discussed using `gp` disks over `io`, but I'm less sure there.

File: config/aggregator/vector.yaml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    type: statsd
+  syslog:
+    address: "[::]:9000"
+    mode: tcp
+    type: syslog
+  vector:
+    address: "[::]:6000"
+    type: vector
+    version: "2"
+sinks:
```
Comment (binarylogic) [inline config/aggregator/vector.yaml:34]: @zsherman @jszwedko what do you think about defaulting with a `pipelines` transform with a couple of commented out pipelines to serve as examples? Without defined pipelines, data would just pass through, so the configuration would still work. The rationale is that we nudge more users towards `pipelines` transform for simplified processing.

File: config/agent/vector.yaml (hunk 12 above)
Comment (spencergilbert) [reply config/agent/vector.yaml:1]: I included them here as they were included in @binarylogic's original commits - but they're not directly related, no.

File: docs/specs/target.md (hunk 11 above)
Comment (spencergilbert) [reply docs/specs/target.md:98]: I think SHOULD is still reasonable - it ought to be done this way unless you understand the implications of not doing it, and have a good reason to ignore the directive. In kubernetes examples it could be binpacking/misc infra tools need to run there/etc.

File: docs/specs/target.md (hunk 10 above)
Comment (spencergilbert) [reply docs/specs/target.md:165]: 👍 should shake out into the get targets to match spec/moving out artifact pipelines onto Datadog

File: docs/specs/target.md (hunk 7 above)
Comment (spencergilbert) [reply docs/specs/target.md:121]: The more I think about this as a spec - I don't think we should be specifically listing machine types, probably just CPU/RAM/DISK(speed/size)

File: docs/specs/target.md (hunk 8 above)
Comment (spencergilbert) [reply docs/specs/target.md:125]: https://github.com/vectordotdev/vector/pull/11477#discussion_r847344953

File: docs/specs/target.md (hunk 9 above)
Comment (spencergilbert) [reply docs/specs/target.md:123]: See: https://github.com/vectordotdev/vector/pull/11477#discussion_r847344953

File: config/aggregator/vector.yaml (hunk 13 above)
Comment (binarylogic) [reply config/aggregator/vector.yaml:34]: @spencergilbert spoke to @zsherman about this. Let's add a `pipelines` transform that all sources fan into. It can be empty for now. I'll think of default pipelines that we can add or comment.

Comment (spencergilbert) [reply config/aggregator/vector.yaml:34]: 👍 I'll figure out pipeline transforms work 😆

Comment (spencergilbert) [reply config/aggregator/vector.yaml:34]: @binarylogic @zsherman, talking with @jszwedko - since `pipelines` haven't been released, is it appropriate to have as a part of our default config?

Comment (jszwedko) [reply config/aggregator/vector.yaml:34]: Yeah, agreed, I think we should leave out pipelines until they are released and documented.

File: config/agent/vector.yaml (hunk 12 above)
Comment (jszwedko) [reply config/agent/vector.yaml:1]: Ah, yeah, I'd split them out and just leave this PR as introducing the spec.

File: docs/specs/target.md (hunk 7 above)
Comment (jszwedko) [reply docs/specs/target.md:121]: Yeah, I think that'd be fine too.

File: docs/specs/target.md (hunk 9 above)
Comment (spencergilbert) [reply docs/specs/target.md]: Removed in 67ae0d94bffcf0797b17497401e4d7ea2ccc6885

File: config/agent/vector.yaml (hunk 12 above)
Comment (spencergilbert) [reply config/agent/vector.yaml]: Removed in 0e17884cc69a9f1cd67890cd573bcf1780559227

File: config/aggregator/vector.yaml (hunk 13 above)
Comment (spencergilbert) [reply config/aggregator/vector.yaml]: Removed the defaults from this PR based on discussion with @jszwedko

File: docs/specs/target.md [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+  [aggregator architecture requirements](#aggregator-architecture).
+- Targets MAY support the [unified architecture][unified_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [unified architecture requirements](#unified-architecture).
+
+### 4. Agent Architecture
+
+The [agent architecture][agent_architecture] deploys Vector on each individual
+node for local data collection. Along with general [hardening](#7-hardening)
+requirements, the following requirements define support for this architecture:
```
Comment (binarylogic) [inline docs/specs/target.md:75]:
```suggestion
The [agent architecture][agent_architecture] deploys Vector on each individual
node for distributed data collection and processing. Along with general
[hardening](#7-hardening) requirements, the following requirements define support
for this architecture:
```

File: docs/specs/target.md [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+
+- Architecture
+  - MUST deploy as a daemon on existing nodes, one Vector process per node.
```
Comment (binarylogic) [inline docs/specs/target.md:78]:
```suggestion
  - MUST deploy as a daemon on existing nodes, not a sidecar -- one Vector process per physical node.
```

File: docs/specs/target.md [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+  - MUST deploy with Vector's [default agent configuration][default_agent_configuration]
+    which largely covers the agent architecture
+    [design recommendations][agent_architecture].
```
Comment (binarylogic) [inline docs/specs/target.md:81]:
```suggestion
  - MUST deploy with Vector's [default agent configuration][default_agent_configuration].
```

File: docs/specs/target.md [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+- Sizing
+  - MUST deploy as a good infrastructure citizen, giving resource priority to
+    other services on the same node.
+  - SHOULD be limited to 2 vCPUs, MUST be overridable by the user.
```
Comment (binarylogic) [inline docs/specs/target.md:85]:
```suggestion
  - SHOULD be limited to 1 vCPUs by default, MUST be overridable by the user.
```

File: docs/specs/target.md [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+  - SHOULD be limited to 2 GiB of memory per vCPU (4 GiB in this case), MUST be
+    overridable by the user.
```
Comment (binarylogic) [inline docs/specs/target.md:87]:
```suggestion
  - SHOULD be limited to 2 GiB of memory per vCPU by default, MUST be
    overridable by the user.
```

File: docs/specs/target.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+### 5. Aggregator Architecture
+
+The [aggregator architecture][aggregator_architecture] deploys Vector onto
+dedicated nodes for data aggregation. Along with general [hardening](#7-hardening)
+requirements, the following requirements define support for this architecture:
+
+- Architecture
+  - SHOULD deploy as a service on dedicated nodes, Vector is the only service
+    on the node.
+  - MUST deploy with a persistent disk that is available between deployments.
```
Comment (binarylogic) [inline docs/specs/target.md:99]:
```suggestion
  - SHOULD deploy with a persistent disk that is available between deployments by default,
     MUST be overridable by the user if they do not want a persistent disk.
```

File: docs/specs/target.md [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+  - Configured Vector ports, including non-default user configured ports,
+    SHOULD be automatically accessible within the Cluster or VPC.
+  - Configured Vector sources, including non-default user configured sources,
+    SHOULD be automatically discoverable via target service discovery
+    mechanisms.
+- Sizing
+  - MUST deploy in a way that takes full advantage of all system resources.
+  - The Vector service SHOULD NOT be artificially limited with resource
+    limiters such as cgroups.
+  - SHOULD request 8 vCPUs, MUST be overridable by the user.
```
Comment (binarylogic) [inline docs/specs/target.md:114]:
```suggestion
  - SHOULD require 8 vCPUs by default, MUST be overridable by the user.
```

File: docs/specs/target.md [hunk 21]
```diff
@@ -0,0 +1,181 @@
...
+  - SHOULD request 2 GiB of memory per vCPU (16 GiB in this case), MUST be
+    overridable by the user.
+  - SHOULD request 6 GiB of disk space per vCPU (48 GiB in this case), MUST be
+    overridable by the user.
```
Comment (binarylogic) [inline docs/specs/target.md:118]:
```suggestion
  - SHOULD request 36 GiB of disk space per vCPU by default (288 GiB in this case),
     MUST be overridable by the user.
```

File: docs/specs/target.md [hunk 22]
```diff
@@ -0,0 +1,181 @@
...
+
+The following are additional requirements for Orchestration Platform installation
+targets:
+
+- High Availability
+  - SHOULD deploy across multiple nodes, MUST be overridable by the user.
+  - SHOULD deploy across multiple availability zones, MUST be overridable by the user.
+- Scaling
+  - MUST deploy along with a load balancer to enable horizontal autoscaling.
```
Comment (binarylogic) [inline docs/specs/target.md:127]:
```suggestion
  - SHOULD provide facilities for provisioning a load balancer to enable horizontal scaling
     out of the box. MUST be overridable by the user.
     - Cloud-managed load balancers (i.e., AWS NLB) SHOULD be supported in addition to
        self-managed load balancers (i.e., HAProxy).
     - Cloud-managed load balancers (i.e., AWS NLB) SHOULD be prioritized by default over
        self-managed load balancers (i.e., HAProxy).
     - Network load balancers (layer-4) SHOULD be prioritized over HTTP load balancers (layer-7)
```

**Review by binarylogic — COMMENTED** [review]
Some changes related to the new docs.

File: docs/specs/target.md [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+  - SHOULD deploy within one Cluster or VPC at a time.
+  - Configured Vector ports, including non-default user configured ports,
+    SHOULD be automatically accessible within the Cluster or VPC.
+  - Configured Vector sources, including non-default user configured sources,
+    SHOULD be automatically discoverable via target service discovery
+    mechanisms.
+- Sizing
+  - MUST deploy in a way that takes full advantage of all system resources.
+  - The Vector service SHOULD NOT be artificially limited with resource
+    limiters such as cgroups.
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply docs/specs/target.md:111]: We can set a `request` that ensures we're on a node with _at least_ that much resource available, and not set a `limit` (which handles throttling/oomkilling Vector)

[the remark this answers is not in the document]
Comment (jszwedko) [reply docs/specs/target.md:111]: Indeed, I figured systemd would make use of cgroups too.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply docs/specs/target.md:109]: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ documents detailing how this works in Kube land

File: docs/specs/target.md [hunk 24]
```diff
@@ -0,0 +1,182 @@
...
+## 2. Installation Targets
+
+Vector supports a number of installation targets that can be categorized into:
+
+- Virtual/Physical Machine
+- Orchestration Platform
+
+The primary differentiator between the two being that Virtual/Physical Machines
+provide a single node as the deployment target, whereas Orchestration Platforms
+allow for a scheduler to deploy Vector across a number nodes. These categories
```
Comment (StephenWakely) [inline docs/specs/target.md:41]:
```suggestion
allow for a scheduler to deploy Vector across a number of nodes. These categories
```

File: docs/specs/target.md [hunk 25]
```diff
@@ -0,0 +1,181 @@
...
+  - SHOULD deploy as a service on dedicated nodes, Vector is the only service
+    on the node.
+  - MUST deploy with a persistent disk that is available between deployments.
+  - MUST deploy with Vector's [default aggregator configuration][default_aggregator_configuration]
+    which largely covers the aggregator architecture
+    [design recommendations][aggregator_architecture].
+    in order to achieve durability with disk buffers and source checkpoints.
+  - SHOULD deploy within one Cluster or VPC at a time.
+  - Configured Vector ports, including non-default user configured ports,
+    SHOULD be automatically accessible within the Cluster or VPC.
```
[the remark this answers is not in the document]
Comment (binarylogic) [reply docs/specs/target.md:101]: That's pretty cool. I'm a fan of it if the complexity doesn't outweigh the benefit.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply docs/specs/target.md:101]: 👍 it _is_ pretty cool if I do say so myself 😆 Being able to do it where possible, but leaving an escape hatch for users seems reasonable to me. There's definitely not a lot of places we'd be able to do it.

File: docs/specs/target.md [hunk 26]
```diff
@@ -0,0 +1,182 @@
...
+    CPU utilization and a stabilization period of 5 minutes.
+
+### 6. Unified Architecture
+
+The [unified architecture][unified_architecture] deploys Vector on each
+individual node for local data collection, as well as onto dedicated nodes for
+data aggregation. The requirements for both the [agent](#4-agent-architecture)
+and the [aggregator](#5-aggregator-architecture) apply to this architecture.
+This architecture SHOULD NOT be installed on Virtual/Physical Machine targets
+as there is little added benefit.
```
Comment (binarylogic) [inline docs/specs/target.md:140]:
```suggestion
The [unified architecture][unified_architecture] deploys Vector on each
individual node as an agent and as a separate service as an aggregator.
The requirements for both the [agent](#4-agent-architecture) and the
[aggregator](#5-aggregator-architecture) apply to this architecture.
This architecture SHOULD NOT be installed on Virtual/Physical Machine
targets as there is little added benefit.
```

**Review by binarylogic — COMMENTED** [review]
I can't approve my own pull request, but approved on my end. Nice work.

**Review by spencergilbert — APPROVED** [review]
Approving on behalf of @binarylogic but it'd be nice to get one from the not authors of this 😆

**Comment by binarylogic** [comment]
@silversupreme would love a review from you since a lot of this reflects the feedback you've received.

File: docs/specs/target.md [hunk 27]
```diff
@@ -0,0 +1,182 @@
...
+  - [5. Aggregator Architecture](#5-aggregator-architecture)
+  - [6. Unified Architecture](#6-unified-architecture)
+- [7. Hardening](#7-hardening)
+
+## 1. Introduction
+
+In its simplest form, installing Vector consists of downloading the binary and
+making it executable, but leaves much to be desired for users looking to
+integrate Vector in real-world production environments. To adhere with Vector's
+["reduce decisions" design principle], Vector must also be opinionated about how
```
Comment (silversupreme) [inline docs/specs/target.md:26]: > ["reduce decisions" design principle]

Was this intended to be a link somewhere?

File: docs/specs/target.md [hunk 28]
```diff
@@ -0,0 +1,182 @@
...
+    mechanisms.
+- Sizing
+  - MUST have dedicated/reserved resources that cannot be stolen by other services, preventing
+     the "noisy neighbor" problem to the degree possible.
+  - The Vector service SHOULD NOT be artificially limited with resource
+    limiters such as cgroups.
+  - SHOULD require 8 vCPUs by default, MUST be overridable by the user.
+  - SHOULD require 2 GiB of memory per vCPU (16 GiB in this case) by default,
+     MUST be overridable by the user.
+  - SHOULD request 36 GiB of disk space per vCPU by default (288 GiB in this case),
```
Comment (silversupreme) [inline docs/specs/target.md:113]: Real glad for clarification on this one, I'll update my POV doc to reflect this.

**Review by silversupreme — APPROVED** [review]
Looks good, just some nits

File: docs/specs/target.md (hunk 27 above)
Comment (spencergilbert) [reply docs/specs/target.md:26]: Oops that got dropped between commits somehow - was intended to link to https://github.com/vectordotdev/vector/blob/master/docs/USER_EXPERIENCE_DESIGN.md#be-opinionated--reduce-decisions

Comment (spencergilbert) [reply docs/specs/target.md]: Fixed in f3f68eec90bbb4568db9023c733ee8c47e368ec7
